internal/assets/token_approval.go

```diff
@@ -107,20 +109,28 @@ func (s *approveSender) resolve(ctx context.Context) (opResubmitted bool, err er
 	if err != nil {
 		// Check if we've clashed on idempotency key. There might be operations still in "Initialized" state that need
 		// submitting to their handlers
+		resubmitWholeTX := false
 		if idemErr, ok := err.(*sqlcommon.IdempotencyError); ok {
-			resubmitted, resubmitErr := s.mgr.operations.ResubmitOperations(ctx, idemErr.ExistingTXID)
+			total, resubmitted, resubmitErr := s.mgr.operations.ResubmitOperations(ctx, idemErr.ExistingTXID)
 			if resubmitErr != nil {
 				// Error doing resubmit, return the new error
 				return false, resubmitErr
 			}
-			if len(resubmitted) > 0 {
+			if total == 0 {
```

Comment:
This logic feels slightly confusing to me. I guess what we're trying to differentiate is these 3 cases:
* no operations have ever been started for this transaction
* some operations were previously initialized for this transaction, but we think they weren't fully submitted, so we've now retried them
* some operations were previously initialized for this transaction, but we think they're all currently in flight, so we can't do anything but continue waiting

We're expressing these 3 states with a combination of two integer returns... wonder if there's a cleaner way to express it?

Reply:
Right, point taken - it may not be worth continuing to iterate on this interface until we have the cycles to do a larger refactor. We should capture that next step as a task though.

Reply:
Raised https://github.com/hyperledger/firefly/issues/1410